macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +271,355 @@ open class Button: NSButton {
 			layer.contentsScale = scale
 		}
 	}
-	
-	/// Tint color of the button content.
-	@objc override public var contentTintColor: NSColor? {
+
+	private var format: ButtonFormat {
 		get {
-			if style == .primaryFilled {
-				return .white
-			} else {
-				return primaryColor
-			}
+			return ButtonFormat(
+				size: self.size,
+				style: self.style,
+				accentColor: self.accentColor
+			)
 		}
 		set {
-			super.contentTintColor = newValue
-			needsDisplay = true
```

Comment:
Do we not need a `needsDisplay = true` here? I'm guessing whoever sets format calls it themselves?

Reply:
Are there any conventions for when to set `needsDisplay = true`?  Would someone setting the `accentColor`, `size`, `style` or `format` expect to need to do it themselves, or would they expect the setting of the property to trigger a display update automatically?  Would this also apply to the properties such as `contentTintColorRest`?

Reply:
I've been operating under the assumptions that 

1- there isn't a significant perf impact for calling `needsDisplay = true`  as it just tells the run loop to re-render the control the next time it runs (regardless if you called it 1 or 100 times)
2- Any change to the UI of the button (AKA all of the properties you mentioned) should have a `needsDisplay = true` call, and that is done by us and not the client. 

In practice, I found that things rendered fine without explicitly calling it, but I defaulted on calling it anyway. @anandrajeswaran or @markavitale perhaps one of my two assumptions is wrong and we don't really need to call it everywhere?

Reply:
I had the same assumption that you could call it whenever and rely on only one coalesced call to the actual display code happening. Pretty sure you can rely on that.

Regarding when to call it, I think it should only be called when we change something that we use in a display override (drawRect: and friends). If we're just setting a property on the button that NSButton uses internally  and we don't use at all, we should be able to trust that the implementation of NSButton will properly mark itself as needing to be drawn.

For something like size, it seems like that would affect layout rather than display so that would need to invalidate the intrinsicContentSize?

Reply:
Added needsDisplay to few more places.  Making most of the colors private helped with this too.

Reply:
@anandrajeswaran how do we invalidate intrinsicContentSize?

Reply:
https://developer.apple.com/documentation/appkit/nsview/1526864-invalidateintrinsiccontentsize

Reply:
Added this invalidation to the size property's didSet.